config.default.json

```diff
@@ -25,10 +25,30 @@
     },
     "apprisk": {
       "validations": [
+        {
+          "url": "https://$CHECKMARX/cxrestapi/auth/identity/connect/token",
+          "body": {
+            "BROKER_VAR_SUB": ["username", "password"],
+            "username": "${CHECKMARX_USERNAME}",
+            "password": "{CHECKMARX_PASSWORD}",
+            "grant_type": "password",
+            "scope": "sast_rest_api",
+            "client_id": "resource_owner_client",
+            "client_secret": "014DF517-39D1-4453-B7B3-9930C563627C"
+          },
+          "method": "post",
+          "headers": {
+            "Content-Type": "application/x-www-form-urlencoded"
```

Comment:
i think we will need to modify `prepareRequestFromFilterResult` to support conversion from json to `URLSearchParams` to support `x-www-form-urlencoded`

Reply:
I am not sure I understand this `support conversion from json to URLSearchParams`. What JSON are you converting into query strings?

Reply:
in order to get checkmarx token we need to send POST request using `application/x-www-form-urlencoded` and inject the user and pass using the `BROKER_VAR_SUB`.
currently the function that replace the Variable substitution sends only JSON body and for this request i need to modify the body as `URLSearchParams`